src/client/datascience/jupyter/kernels/kernelSelections.ts

```diff
@@ -315,19 +380,45 @@ export class KernelSelectionProvider {
-                                (this.fs.areLocalPathsSame(
-                                    (installedKernel.selection.kernelSpec?.argv || [])[0],
-                                    item.selection.interpreter?.path || ''
+                        installedKernels.find((installedKernel) => {
+                            if (!isPythonKernelConnection(installedKernel.selection)) {
+                                return false;
+                            }
+
+                            const kernelDisplayName =
+                                installedKernel.selection.kernelSpec?.display_name ||
+                                installedKernel.selection.kernelSpec?.name ||
+                                '';
+                            // Possible user has a kernel named `Python` or `Python 3`.
+                            // & if we have such a kernel, we should not display the corresponding interpreter.
+                            if (
+                                kernelDisplayName !== item.selection.interpreter?.displayName &&
+                                !isSimplePythonDisplayName.test(kernelDisplayName.toLowerCase())
+                            ) {
+                                return false;
+                            }
+
+                            // If the python kernel belongs to an existing interpreter with the same path,
+                            // Or if the python kernel has the exact same path as the interpreter, then its a duplicate.
+                            // Paths on windows can either contain \ or / Both work.
+                            // Thus, C:\Python.exe is the same as C:/Python.exe
+                            // In the kernelspec.json we could have paths in argv such as C:\\Python.exe or C:/Python.exe.
+                            const interpreterPathToCheck = (item.selection.interpreter.path || '').replace(/\\/g, '/');
+                            return (
+                                this.fs.areLocalPathsSame(
+                                    ((installedKernel.selection.kernelSpec?.argv || [])[0] || '').replace(/\\/g, '/'),
+                                    interpreterPathToCheck
                                 ) ||
-                                    this.fs.areLocalPathsSame(
-                                        installedKernel.selection.kernelSpec?.metadata?.interpreter?.path || '',
-                                        item.selection.interpreter?.path || ''
-                                    ))
-                        )
+                                this.fs.areLocalPathsSame(
+                                    (
+                                        installedKernel.selection.kernelSpec?.interpreterPath ||
+                                        installedKernel.selection.kernelSpec?.metadata?.interpreter?.path ||
+                                        ''
+                                    ).replace(/\\/g, '/'),
```

Comment:
@IanMatthewHuff @rchiodo 
Added this change to account for duplicates seen in https://github.com/microsoft/vscode-jupyter/issues/399
Not sure this is the cause of the duplicates, but would like to fix this as well
Will need to test. Fortunately Claudia can replicate this.